build_library/build_image_util.sh

```diff
@@ -782,18 +779,70 @@ EOF
         "${BUILD_DIR}/${image_kconfig}"
   fi
 
+  # Build the selinux policy
+  if pkg_use_enabled coreos-base/coreos selinux; then
+      sudo chroot "${root_fs_dir}" bash -c "cd /usr/share/selinux/mcs && semodule -s mcs -i *.pp"
+  fi
+
+  # Run tmpfiles once to make sure that /etc has everything in place before
+  # we freeze it in /usr/share/flatcar/etc as lowerdir in the overlayfs.
+
+  # But first, to successfully run tmpfiles, we need to have all users/groups
+  # in /etc/passwd, and afterwards we can recreate the files for the dev
+  # container with flatcar-tmpfiles (not really needed but maybe nice to have
+  # as it also lands as reference in /usr/share/flatcar/etc).
+  local dbfile
+  for dbfile in passwd shadow group gshadow; do
+    sudo cp -f "${root_fs_dir}"/usr/share/baselayout/"${dbfile}" "${root_fs_dir}"/etc/
+  done
+  sudo systemd-tmpfiles --create --remove --boot --exclude-prefix=/dev --root="${root_fs_dir}"
+  for dbfile in passwd shadow group gshadow; do
+    sudo rm -f "${root_fs_dir}"/etc/"${dbfile}"
+  done
+  sudo "${root_fs_dir}"/usr/sbin/flatcar-tmpfiles "${root_fs_dir}"
+  # Now that we used the tmpfiles for creating /etc we delete them because
+  # the L, d, and C entries cause upcopies
+  sudo sed -i '/^[CLd] *\/etc\//d' "${root_fs_dir}"/usr/lib/tmpfiles.d/*
+
+  # SELinux: Label the root filesystem for using 'file_contexts'.
+  # The labeling has to be done before moving /etc to /usr/share/flatcar/etc to prevent wrong labels for these files and as
+  # the relabeling on boot would cause upcopies in the overlay.
+  # TODO: Breaks the system:
+  # sudo setfiles -Dv -r "${root_fs_dir}" "${root_fs_dir}"/etc/selinux/mcs/contexts/files/file_contexts "${root_fs_dir}"
+  # sudo setfiles -Dv -r "${root_fs_dir}" "${root_fs_dir}"/etc/selinux/mcs/contexts/files/file_contexts "${root_fs_dir}"/usr
+  # For now we only try it with /etc
+  sudo setfiles -Dv -r "${root_fs_dir}" "${root_fs_dir}"/etc/selinux/mcs/contexts/files/file_contexts "${root_fs_dir}"/etc
```

Comment:
What about the case where selinux is disabled?
Wouldn't it better to have `if pkg_use_enabled coreos-base/coreos selinux; then` for the selinux-related commands?

Reply:
Thanks, will use it here, too!